### ISSUE TITLE
Implement JAX conversion for Eye Op

### ISSUE THREAD

**Issue by krishvishal** [pymc-devs/Theano-PyMC#121]
If you have any questions, please ask the [theano-user mailing list](https://groups.google.com/forum/#!forum/theano-users) or [stackoverflow](http://stackoverflow.com/) (using the "theano" tag) first.

## Description of your problem or feature request
I'm following the [notebook](https://gist.github.com/junpenglao/fe5e1b451c076cc7b4ca16acdd7d6472) to make use of JAX and Tensorflow probability for inference of pymc3 model. The following is the code for my model.

**Please provide a minimal, self-contained, and reproducible example.**
```python
y = np.random.rand(10,1)
A = np.log(np.random.rand(10,10))
m,n= np.shape(A)
mc = np.shape(y)[1]

with pm.Model() as model:
    q = pm.Uniform('scale', 4, 5)
    lsd = pm.Uniform('lsd', 1, 2, shape=n)
    u = pm.Bound(pm.Laplace, lower=1, upper=10)('u', mu=5.0, b=lsd, shape=n) 
    mu = pmmath.logsumexp(A - u / q, axis=0)
    noise_sigma = pm.Bound(pm.Normal, lower=0)('noise_sigma', mu=0, sigma=1)
    y_rv = pm.MvNormal('y', mu=tt.tile(mu, (mc,1)).T, chol=noise_sigma * tt.eye(mc), shape=(m,mc), observed=y)
```

The model runs without errors but when I jax compile it, I get

```
TypeError: 'list' object is not callable
```

**Please provide the full traceback of any errors.**
https://pastebin.com/yL85rCL1

**Please provide any additional information below.**

## Versions and main components

* Theano version: 1.0.9
* Theano config (`python -c "import theano; print(theano.config)"`)
* Python version: 3.8.5
* Operating system: Ubuntu 18.04
* How did you install Theano: install from source

@junpenglao

**Comment by brandonwillard** [comment]
The notebook recommends that one install the old `jax-linker` branch.  That branch was merged and is now very far behind.

Assuming this is the MWE you're running, under the version of Theano-PyMC, I get the following:
```python
import jax
import numpy as np

import pymc3 as pm
import theano
import theano.tensor as tt
from theano.sandbox.jaxify import jax_funcify

# Disable C compilation by default
theano.config.cxx = ""

# This will make the JAX Linker the default
theano.config.mode = "JAX"


y = np.random.rand(10, 1)
A = np.log(np.random.rand(10, 10))
m, n = np.shape(A)
mc = np.shape(y)[1]

with pm.Model() as model:
    q = pm.Uniform("scale", 4, 5)
    lsd = pm.Uniform("lsd", 1, 2, shape=n)
    u = pm.Bound(pm.Laplace, lower=1, upper=10)("u", mu=5.0, b=lsd, shape=n)
    mu = pm.math.logsumexp(A - u / q, axis=0)
    noise_sigma = pm.Bound(pm.Normal, lower=0)("noise_sigma", mu=0, sigma=1)
    y_rv = pm.MvNormal(
        "y",
        mu=tt.tile(mu, (mc, 1)).T,
        chol=noise_sigma * tt.eye(mc),
        shape=(m, mc),
        observed=y,
    )


fgraph = theano.gof.FunctionGraph(model.free_RVs, [model.logpt])
fns = jax_funcify(fgraph)
logp_fn_jax = fns[0]

rv_names = [rv.name for rv in model.free_RVs]
init_state = [model.test_point[rv_name] for rv_name in rv_names]
nchain = 5
init_state_batched = jax.tree_map(
    lambda x: np.repeat(x[None, ...], nchain, axis=0), init_state
)

dlogp_logp_fn = jax.value_and_grad(logp_fn_jax, argnums=range(len(init_state)))
jax.vmap(dlogp_logp_fn)(*init_state_batched)
```
```python
NotImplementedError: No JAX conversion for the given `Op`: Eye{dtype='float64'}
```

**Comment by canyon289** [comment]
Thanks Thomas!

**Comment by eigenfoo** [comment]
Closed by #153.
